### PR TITLE
scs 2.1.1 (new formula)

### PR DESCRIPTION
**Pull request by jamiesnape**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/scs.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        const int result = scs(data, cone, solution, &info);
+        scs_free_data(data, cone); scs_free_sol(solution);
+        return result - SCS_SOLVED;
+      }
+    EOS
+    system ENV.cc, "test.c", "-I#{include}/scs", "-L#{lib}", "-lscsindir",
+      "-otestscsindir"
+    system "./testscsindir"
+    system ENV.cc, "test.c", "-I#{include}/scs", "-L#{lib}", "-lscsdir",
+      "-otestscsdir"
```
Comment (fxcoudert) [inline Formula/scs.rb]: Same here

File: Formula/scs.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+class Scs < Formula
+  desc "Conic optimization via operator splitting"
+  homepage "https://web.stanford.edu/~boyd/papers/scs.html"
+  url "https://github.com/cvxgrp/scs/archive/2.1.1.tar.gz"
+  sha256 "0e20b91e8caf744b84aa985ba4e98cc7235ee33612b2bad2bf31ea5ad4e07d93"
+
+  resource "scs-test-data-small-random-socp" do
```
Comment (fxcoudert) [inline Formula/scs.rb]: Is this part of the release tarball? If so, we could install the file as part of the `install`:

```
pkgshare.install "test/data/small_random_socp"
```

so we don't have to download it again for tests

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/scs.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        const int status = scs_read_data("small_random_socp", &data, &cone);
+        ScsSolution *solution = scs_calloc(1, sizeof(ScsSolution));
+        ScsInfo info;
+        const int result = scs(data, cone, solution, &info);
+        scs_free_data(data, cone); scs_free_sol(solution);
+        return result - SCS_SOLVED;
+      }
+    EOS
+    system ENV.cc, "test.c", "-I#{include}/scs", "-L#{lib}", "-lscsindir",
+      "-otestscsindir"
```
[the remark this answers is not in the document]
Comment (jamiesnape) [reply Formula/scs.rb]: Done.

File: Formula/scs.rb (hunk 2 above)
Comment (jamiesnape) [reply Formula/scs.rb]: Done.

File: Formula/scs.rb (hunk 1 above)
Comment (jamiesnape) [reply Formula/scs.rb]: Done.

Comment (jamiesnape) [reply Formula/scs.rb]: Done.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thank you @jamiesnape

**Pull request closed without merging**